allennlp/sanity_checks/task_checklists/task_suite.py

```diff
@@ -0,0 +1,293 @@
+        **kwargs : `type`
+            Will be passed as arguments to each test.summary()
+        """
+        old_stdout = sys.stdout
+        try:
+            sys.stdout = file
+            self.suite.summary(capabilities=capabilities, **kwargs)
+        finally:
+            sys.stdout = old_stdout
+
+    def run(
+        self,
+        predictor: Predictor,
+        capabilities: Optional[List[str]] = None,
+        max_examples: Optional[int] = None,
+    ):
+        """
+        Runs the predictor on the test suite data.
+
+        # Parameters
+
+        predictor : `Predictor`
+            The predictor object.
+        capabilities : `List[str]`, optional (default = `None`)
+            If not None, will only run tests with these capabilities.
+        max_examples : `int`, optional (default = `None`)
+            Maximum number of examples to run. If None, all examples will be run.
+        """
+        preds_and_confs_fn = self._prediction_and_confidence_scores(predictor)
+        if preds_and_confs_fn is NotImplementedError:
+            raise NotImplementedError(
+                "The `_prediction_and_confidence_scores` function needs "
+                "to be implemented for the class `{}`".format(self.__class__)
+            )
+        if not capabilities:
+            self.suite.run(preds_and_confs_fn, overwrite=True, n=max_examples)
+        else:
+            for _, test in self.suite.tests.items():
+                if test.capability in capabilities:
+                    test.run(preds_and_confs_fn, verbose=True, overwrite=True, n=max_examples)
+
+    @classmethod
+    def constructor(
+        cls,
+        name: Optional[str] = None,
+        suite_file: Optional[str] = None,
+        extra_args: Optional[Dict[str, Any]] = None,
+    ) -> "TaskSuite":
+        suite_class: Type[TaskSuite] = (
+            TaskSuite.by_name(name) if name is not None else cls  # type: ignore
+        )
+
+        if extra_args is None:
+            extra_args = {}
+
+        if suite_file is not None:
+            return suite_class(TestSuite.from_file(cached_path(suite_file)), **extra_args)
+        return suite_class(**extra_args)
+
+    def save_suite(self, suite_file: str):
+        """
+        Saves the suite to a file.
+        """
+        self.suite.save(suite_file)
+
+    def _default_tests(self, data: Optional[Iterable], num_test_cases=100):
+        """
+        Derived TaskSuite classes can add any task-specific tests here.
+        """
+        if data:
+
+            # Robustness
+
+            self._punctuation_test(data, num_test_cases)
+            self._typo_test(data, num_test_cases)
+            self._contraction_test(data, num_test_cases)
+
+    @classmethod
+    def contractions(cls):
+        return Perturb.contractions
+
+    @classmethod
+    def typos(cls):
+        return Perturb.add_typos
+
+    @classmethod
+    def punctuation(cls):
+        return utils.toggle_punctuation
```

Comment:
You renamed this to `strip_punctuation` elsewhere, didn't you?

Reply:
No, there are two separate functions. `strip_punctuation` only removes the punctuation. `toggle_punctuation` will also add punctuation if the input doesn't contain it.